docs/conflict_resolution.md

```diff
@@ -0,0 +1,81 @@
+- You can resolve conflicts based on your business logic and again make a call to the server.
+
+### Code:
+
+```kotlin
+/*
+UserConflictResolutionHandler extends ConfliceResolutionInterface.
+Here the user provides the custom implementation of resolving conflicts.
+ */
+class UserConflictResolutionHandler(val context: Context) : ConfliceResolutionInterface {
+
+    /*
+    Function which resolve the conflicts based on the user business logic.
+     */
+    override fun resolveConflict(
+        serverState: Map<String, Any>,
+        clientState: Map<String, Any>,
+        operationType: String
+    ) {
+        /*
+        Version based approach of Conflict Resolution.
+        */
+        val serverMap = serverState
+        val containsVersion = serverMap.containsKey("version")
+
+        if (containsVersion) {
+            var versionAfterConflict = serverMap.get("version") as Int + 1
+
+        /* You can run a switch case on the operation type to detect which type of mutation is it in which conflict occured
+           and accordingly create an object of that mutation, resolve conflict and make a server call with it.
+         */  
+            when (operationType) {
+                /*
+                According to the schema structure, used a version based approach of resolving conflicts.
+                If operationType is "UpdateCurrentTaskMutation" perform the following steps to resolve conflicts.
+                 */
+                "UpdateCurrentTaskMutation" -> {
+                    var mut = UpdateCurrentTaskMutation.builder().id(clientState["id"].toString())
+                        .title(clientState["title"].toString()).version(
+                            versionAfterConflict
+                        ).build()
+
+                    val customCallback = object : ResponseCallback {
+
+                        override fun onSuccess(response: Response<Any>) {
+                            Log.e("onSuccess() updateTask", "${response.data()}")
+                            val result = response.data()
+
+                            //In case of conflicts data returned from the server id null.
+                            result?.let {
+                                Log.e(TAG, "onResponse-UpdateTask- $it")
+                            }
+                        }
+
+                        override fun onSchedule(
+                            e: ApolloException,
+                            mutation: Mutation<Operation.Data, Any, Operation.Variables>
+                        ) {
+                            Log.e("onSchedule() updateTask", "${mutation.variables().valueMap()}")
+                            e.printStackTrace()
+                        }
+                    }
+
+                    Utils.getApolloClient(context)?.enqueue(
```

Comment:
Utils name is very common. This is end user api so if we can provide bette name like

`OffixHelpers`?

Reply:
Yeah, removed from here and mentioned simple apollo client in the docs.